### ISSUE TITLE
Why are relations nullified when all values are null?

### ISSUE THREAD

**Issue by daffodil11** [typeorm/typeorm#5177]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

Our use case has an entity for which all columns except the primary key are nullable. This entity is used as a relation, and it is valid for all the values in this entity to be null. When this is the case, we expect the relation to be an object with all the properties set to null. Instead, the _relation itself_ is null.

For example, if we get an entity which has an id column and a relation called details, and details has name and message columns with values of null.

Expected:
```javascript
{
  id: 1,
  details: {
    name: null,
    message: null,
  }
}
```

Actual:
```javascript
{
  id: 1,
  details: null
}
```

We have identified the line in the codebase that causes this behaviour: https://github.com/typeorm/typeorm/blob/fb6068849215e6ea75222f7bb015030314cf67d6/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts#L144

The comment makes it clear that the this is not a bug - the behaviour we have observed is a result of this line working as intended.

So, our question is: What are the reasons for the decision to return an object as null if all its values are null?

**Comment by fan-tom** [comment]
Do you have primary key in `details` relation? If no, how can you link to it? If yes, can you update question to clarify that, i.e

expected
```javascript
{
  id: 1,
  details: {
    id: 5,
    name: null,
    message: null,
  }
}
```

actual
```javascript
{
  id: 1,
  details: null
}
```

**Comment by daffodil11** [comment]
The parent entity (let's call it "Post" for brevity) has a primary key (id), and the entity used for details (let's call it "Details") has a OneToOne relation to Post, so it has a join column ("post") with the primary option set to true (in the database table, the column would be named "postId"). When using this pattern, it appears that TypeORM omits the primary key ("postId") from Details when Post is loaded from the database.

This is not a problem (and in fact is sensible, given that the primary key of Details is always the same as the primary key of Post, so it would be redundant to include it as a property on both Post and Details).

**Comment by puroong** [comment]
Is there any updates on this issue?

**Comment by imnotjames** [comment]
To answer the question - this is done because of how typeorm loads via joins.  If all of the fields are null then that means the join didn't return any results.

However.. it should have at least gotten the primary key.  What are your entity definitions?

**Comment by imnotjames** [comment]
Duplicate of #4299

**Comment by imnotjames** [comment]
Also fixed via #7450
